packages/fxa-payments-server/src/components/fields/index.tsx

```diff
@@ -113,8 +113,16 @@ export type OnValidateFunction = (
   getString?: Function
 ) => { value: any; valid: boolean | undefined; error: any };
 
+export type OnValidateFunctionPromise = (
+  value: any,
+  focused: boolean,
+  props: FieldProps,
+  getString?: Function
+) => Promise<{ value: any; valid: boolean | undefined; error: any }>;
+
 type InputProps = {
   onValidate?: OnValidateFunction;
+  onValidatePromise?: OnValidateFunctionPromise;
```

Comment:
Added this to avoid breaking functionality across the codebase. This works now, but I wonder if it would be better to have `onValidate` accept both it's current return signature and a promise wrapped version. I tried a couple of different methods but didn't figure it out fully. So took this direction to keep it moving since we have an august deadline. Would be happy to try suggestions here, or create a tech debt/follow up ticket to address it.

Reply:
We ought to clean this up, but im ok with it for now. Can you create a maintenance issue for it?

Reply:
I believe this should work with the return value of the validation function being a sum type, i.e.:

```
export type ValidationResult = {
  value: any;
  valid: boolean | undefined;
  error: any;
};
export type OnValidateFunction = (
  value: any,
  focused: boolean,
  props: FieldProps,
  getString?: Function
) => ValidationResult | Promise<ValidationResult>;
```

And then await all the `onValidate` calls.

Reply:
https://mozilla-hub.atlassian.net/browse/FXA-3802

Reply:
Did my suggestion not work?

Reply:
@chenba I tried that initially before opening the pr and I was still having lots of test failures, so will address this in the follow up issue when we aren't working toward a deadline.